module/VuFind/src/VuFind/Feature/MergeRecursiveTrait.php

```diff
@@ -70,7 +70,7 @@ protected function isStringKeyedArray($op)
      */
     protected function mergeRecursive($val1, $val2)
     {
-        if (empty($val2)) {
+        if ($val2 === null || is_array($val2) && empty($val2)) {
```

Comment:
Would `$val2 === []` be a more efficient way to express `is_array($val2) && empty($val2)`? [The manual](https://www.php.net/manual/en/language.operators.array.php) seems to suggest this should be safe.

Reply:
Yes, that makes sense. I changed it.